pgconnection.js

```
@@ -0,0 +1,100 @@
+          segmentedData[keys[i]].push(data[keys[i]].slice(j - step, j));
+        }
+      }
+    }
+
+    const queryText = format(
+      "UPDATE %I SET %I = $1 WHERE %I = ANY($2::int[]);",
+      ways_table_name,
+      stress_level_column_name, 
+      osm_id_column_name
+    );
+
+    for (let i = options.zero ? 0 : 1; i < keys.length; i++) {
+      if (options.verbose) {
+        console.log(`Updating level ${i} ways.`);
+      }
+      const way_ids = segmentedData[keys[i]];
+
+      for (let j = 0; j < way_ids.length; j++) {
+        const query = {
+          name: 'update-levels',
+          text: queryText,
+          values: [i, way_ids[j]]
+        };
+        await client.query(query);
+      }
+    }
+
+    if (options.verbose) {
+      const nullRes = await client.query(format("SELECT count(*) FROM %I WHERE %I IS NULL", ways_table_name, stress_level_column_name));
+      const zeroRes = await client.query(format("SELECT count(*) FROM %I WHERE %I = 0", ways_table_name, stress_level_column_name));
+      const oneRes = await client.query(format("SELECT count(*) FROM %I WHERE %I = 1", ways_table_name, stress_level_column_name));
+      const twoRes = await client.query(format("SELECT count(*) FROM %I WHERE %I = 2", ways_table_name, stress_level_column_name));
+      const threeRes = await client.query(format("SELECT count(*) FROM %I WHERE %I = 3", ways_table_name, stress_level_column_name));
+      const fourRes = await client.query(format("SELECT count(*) FROM %I WHERE %I = 4", ways_table_name, stress_level_column_name));
+      console.log(
+        `Ways levels count:
+          NULL = ${nullRes.rows[0].count}, 
+          0 = ${zeroRes.rows[0].count}, 
+          1 = ${oneRes.rows[0].count}, 
+          2 = ${twoRes.rows[0].count}, 
+          3 = ${threeRes.rows[0].count}, 
+          4 = ${fourRes.rows[0].count}`
+      );
+    }
+
+    await client.end();
+  
+  } catch (error) {
+    console.error('Database connection error:', error);
```

The program should exit with a non-zero code if it fails to connect to the database, otherwise it could hang a CI script

---

I added a return value and a promise .then to check for it. Should return an error now if it fails